sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTable.scala

```diff
@@ -0,0 +1,129 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.v2.jdbc
+
+import java.util
+
+import scala.collection.JavaConverters._
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.{DataFrame, Row, SparkSession, SQLContext}
+import org.apache.spark.sql.connector.catalog.{Identifier, SupportsRead, SupportsWrite, Table, TableCapability}
+import org.apache.spark.sql.connector.read.{Scan, ScanBuilder, SupportsPushDownFilters, SupportsPushDownRequiredColumns, V1Scan}
+import org.apache.spark.sql.connector.write.{LogicalWriteInfo, SupportsTruncate, V1WriteBuilder, WriteBuilder}
+import org.apache.spark.sql.execution.datasources.jdbc.{JDBCOptions, JdbcOptionsInWrite, JDBCRDD, JDBCRelation, JdbcUtils}
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.jdbc.JdbcDialects
+import org.apache.spark.sql.sources.{BaseRelation, Filter, InsertableRelation, TableScan}
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
+
+case class JDBCTable(
+    ident: Identifier,
+    schema: StructType,
+    jdbcOptions: JDBCOptions) extends Table with SupportsRead with SupportsWrite {
+  assert(ident.namespace().length == 1)
+
+  override def name(): String = ident.toString
+
+  override def capabilities(): util.Set[TableCapability] = {
+    import TableCapability._
+    Set(BATCH_READ, V1_BATCH_WRITE, TRUNCATE).asJava
+  }
+
+  override def newScanBuilder(options: CaseInsensitiveStringMap): ScanBuilder = {
+    val mergedOptions = new JDBCOptions(
+      jdbcOptions.parameters.originalMap ++ options.asCaseSensitiveMap().asScala)
+    val session = SparkSession.active
+    val resolver = session.sessionState.conf.resolver
+    val timeZoneId = session.sessionState.conf.sessionLocalTimeZone
+    val parts = JDBCRelation.columnPartition(schema, resolver, timeZoneId, mergedOptions)
+    new JDBCScan(JDBCRelation(schema, parts, mergedOptions)(session))
+  }
+
+  override def newWriteBuilder(info: LogicalWriteInfo): WriteBuilder = {
+    val mergedOptions = new JdbcOptionsInWrite(
+      jdbcOptions.parameters.originalMap ++ info.options.asCaseSensitiveMap().asScala)
+    new JDBCWrite(schema, mergedOptions)
+  }
+}
+
+class JDBCScan(relation: JDBCRelation) extends ScanBuilder with V1Scan
```

Comment:
Nit: how about moving `JDBCScan` to a single file?

Reply:
Also, it would be great if we split it into JDBCScanBuilder and JDBCScan, for demonstrating the design idea of DSV2 API.